### DISCUSSION TITLE
How to implement newsletter subscription

### DISCUSSION THREAD

**Discussion by sbkjarmul** [vuestorefront/prestashop#27] [Q&A]
Hi everybody,

I would like to create a Newsletter Page for my shop using Prestashop and VueStorefront. I can't find ready composable for that.
Can somebody explain me how it should look like?

I've  already created api file below, which uses appropriate endpoint:

<img width="650" alt="image" src="https://user-images.githubusercontent.com/37276167/163707834-a9e66e5c-ca06-41a3-988e-6a1fcd3fd804.png">

**Comment by samberrry** [comment]
Hi,

you should create a custom composable for newsletter, you can check bootstrap composable to see how to create that. If you are interested in you can create a PR to add it into the main project as a contributor 😍

To do that please, create a branch from develop branch (name it something meaningful) and create your PR. 
Also, for this feature we do not need a separate page, as you see there is a pop up. You can complete it by adding logic behind that pop up. 

thanks
